kf_lib_data_ingest/target_api_plugins/kids_first_dataservice.py

```diff
@@ -278,8 +278,12 @@ def build_entity(record, key, get_target_id_from_record):
         return {
             "kf_id": get_target_id_from_record(Biospecimen, record),
             "participant_id": get_target_id_from_record(Participant, record),
-            "external_sample_id": record.get(CONCEPT.BIOSPECIMEN_GROUP.ID),
-            "external_aliquot_id": record.get(CONCEPT.BIOSPECIMEN.ID),
+            "external_sample_id": record.get(
+                CONCEPT.BIOSPECIMEN_GROUP.ID, record.get(CONCEPT.BIOSPECIMEN.ID)
+            ),
+            "external_aliquot_id": record.get(
+                CONCEPT.BIOSPECIMEN.ID, record.get(CONCEPT.BIOSPECIMEN_GROUP.ID)
+            ),
```

Comment:
I can understand setting group ID from the sample if a group isn't designated, but should we really set the sample from group? I would think that explicitly setting group but not sample suggests that sample should have been set and is missing by accident?

Reply:
Hmm, yea I agree with @fiendish. If group is not set, use sample. If sample is not set, don't do anything